client/smarthomeconnect.cpp

```diff
@@ -0,0 +1,276 @@
+#include "smarthomeconnect.h"
+
+
+namespace smart_home_client
+{
+
+
+void SmartHomeConnect::StartConnection()
+{
+    std::unique_ptr<int, SocketDeleter> clientSocket(new int(socket(AF_INET, SOCK_STREAM, 0)));
+
+    if (*clientSocket == -1)
+    {
+        std::cerr << "Failed to create socket" << std::endl;
+        return;
+    }
+    
+	sockaddr_in serverAddress; 
+	serverAddress.sin_family = AF_INET; 
+	serverAddress.sin_port = htons(8080); 
+	serverAddress.sin_addr.s_addr = INADDR_ANY; 
+
+	if(connect(*clientSocket, (struct sockaddr*)&serverAddress, sizeof(serverAddress)) == -1)
+    {
+        std::cerr << "Failed to bind socket" << std::endl;
+        return;
+    } 
+
+    while (true)
```

Comment:
This is no big deal, but personally I'd move this logic outside the function(maybe into a run function or something similar). SOLID - S, your function should only do one thing: start a connection.(you don't need to change it, just keep in mind)